dataland-frontend/tests/e2e/specs/eu-taxonomy-non-financials/DataUpload.ts

```diff
@@ -116,45 +118,67 @@ describeIf(
       () => {
         getKeycloakToken(admin_name, admin_pw).then((token: string) => {
-          const dummyCompanyInformation = generateDummyCompanyInformation(testData.companyInformation.companyName);
+          const dummyCompanyInformation = generateDummyCompanyInformation(`Company-For-DataUpload-test-${Date.now()}`);
           return uploadCompanyViaApi(token, dummyCompanyInformation).then((storedCompany) => {
             cy.ensureLoggedIn(admin_name, admin_pw);
+
             cy.visitAndCheckAppMount(
               `/companies/${storedCompany.companyId}/frameworks/${DataTypeEnum.EutaxonomyNonFinancials}/upload`,
             );
             uploadReports.selectFile(TEST_PDF_FILE_NAME);
             uploadReports.selectFile(`${TEST_PDF_FILE_NAME}2`);
             uploadReports.fillAllFormsOfReportsSelectedForUpload(2);
-            fillAndValidateEuTaxonomyForNonFinancialsUploadForm(`${TEST_PDF_FILE_NAME}2`);
+            fillRequiredEutaxonomyNonFinancialsFields();
             cy.get('div[name="revenue"]').within(() => {
+              cy.get('[data-test="dataPointToggleButton"]').click();
+              cy.get('input[name="value"]').type("250700");
+              cy.get('select[data-test="datapoint-currency"]').select(1);
+              cy.get('select[name="quality"]').select(1);
               cy.get('select[name="fileName"]').select(TEST_PDF_FILE_NAME);
             });
             cy.get('div[name="capex"]').within(() => {
+              cy.get('[data-test="dataPointToggleButton"]').click();
+              cy.get('input[name="value"]').type("450700");
+              cy.get('select[data-test="datapoint-currency"]').select(10);
+              cy.get('select[name="quality"]').select(1);
               cy.get('select[name="fileName"]').select(`${TEST_PDF_FILE_NAME}2`);
             });
             cy.intercept({ method: "POST", url: `**/api/data/**`, times: 1 }, (request) => {
-              const data = assertDefined(request.body as CompanyAssociatedDataEuTaxonomyDataForNonFinancials).data;
-              expect(TEST_PDF_FILE_NAME in assertDefined(data.general?.referencedReports)).to.equal(true);
-              expect(`${TEST_PDF_FILE_NAME}2` in assertDefined(data.general?.referencedReports)).to.equal(true);
-              frontendDocumentHash = assertDefined(data.general?.referencedReports)[TEST_PDF_FILE_NAME].fileReference;
-            }).as("submittedData");
+              const submittedEutaxonomyNonFinancialsData = assertDefined(
+                request.body as CompanyAssociatedDataEutaxonomyNonFinancialsData,
+              ).data;
+              const submittedReferencedReports = assertDefined(
+                submittedEutaxonomyNonFinancialsData.general?.referencedReports,
+              );
+              expect(`${TEST_PDF_FILE_NAME}2` in submittedReferencedReports).to.equal(true);
+              if (TEST_PDF_FILE_NAME in submittedReferencedReports) {
```

Comment:
Well ist must be.

Reply:
Ich muss dieses if machen, weil typescript in der Folgezeile sonst sagt 

"Woher willst du wissen dass dieser Index-Zugriff hier funktioniert?:"

`submittedReferencedReports[TEST_PDF_FILE_NAME]`

Reply:
Belastend